src/utils/objectUtils.ts

```diff
@@ -175,3 +175,48 @@ export function deepClone<T extends object | undefined | null>(object: T): T {
   }
   return structuredClone(object)
 }
+
+type TransformToKebabCaseInputType = Record<string, unknown> | null | undefined
+type TransformToKebabCaseReturnType<Input, Output> =
+  Input extends Record<string, unknown> ? Output : Input
+/**
+ * Transforms an object's keys from camelCase or snake_case to kebab-case.
+ * @param object
+ */
+export function transformToKebabCase<
+  Output extends Record<string, unknown>,
+  Input extends TransformToKebabCaseInputType,
+>(object: Input): TransformToKebabCaseReturnType<Input, Output>
```

Comment:
I tried to automatically infer the return type,  and it was working for simple objects just with `snake_case` or `camelCase` but was complex(check), after trying to add edge cases (non-alphanumeric + abbreviations + nested arrays) it was even more complex and didn't work as expected, so to avoid weird behaviour I decided to just allow programmer to specify the return type